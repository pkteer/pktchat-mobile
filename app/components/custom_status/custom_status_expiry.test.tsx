// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
// See LICENSE.txt for license information.

import moment from 'moment-timezone';
import React from 'react';

import Preferences from '@mm-redux/constants/preferences';
import {renderWithReduxIntl} from '@test/testing_library';

import CustomStatusExpiry from './custom_status_expiry';

jest.mock('@mm-redux/selectors/entities/preferences');

jest.mock('@mm-redux/selectors/entities/preferences', () => ({
    getBool: jest.fn().mockReturnValue(false),
    isCollapsedThreadsEnabled: jest.fn(),
}));

describe('components/custom_status/custom_status_expiry', () => {
    const date = moment().endOf('day');

    const baseProps = {
        theme: Preferences.THEMES.denim,
        time: date.toDate(),
    };

    it('should match snapshot', () => {
        const wrapper = renderWithReduxIntl(
            <CustomStatusExpiry
                {...baseProps}
            />,
        );
        expect(wrapper.toJSON()).toMatchSnapshot();
        expect(wrapper.getByText('Today')).toBeDefined();
    });

    it('should match snapshot with prefix and brackets', () => {
        const props = {
            ...baseProps,
            showPrefix: true,
            withinBrackets: true,
        };
        const wrapper = renderWithReduxIntl(
            <CustomStatusExpiry
                {...props}
            />,
        );
        expect(wrapper.toJSON()).toMatchSnapshot();
        expect(wrapper.getByText('Until')).toBeDefined();
        expect(wrapper.getByText('Today')).toBeDefined();
    });

    it("should render Tomorrow if given tomorrow's date", () => {
        const props = {
            ...baseProps,
            time: moment().add(1, 'day').endOf('day').toDate(),
        };
        const wrapper = renderWithReduxIntl(
            <CustomStatusExpiry
                {...props}
            />,
        );
        expect(wrapper.getByText('Tomorrow')).toBeDefined();
    });
});
