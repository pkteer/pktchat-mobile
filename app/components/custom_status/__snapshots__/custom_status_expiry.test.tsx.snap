// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/custom_status/custom_status_expiry should match snapshot 1`] = `
<Text
  style={
    Array [
      Object {
        "color": "#3f4350",
        "fontSize": 15,
      },
      Object {},
    ]
  }
  testID=""
>
  <Text>
    Today
  </Text>
</Text>
`;

exports[`components/custom_status/custom_status_expiry should match snapshot with prefix and brackets 1`] = `
<Text
  style={
    Array [
      Object {
        "color": "#3f4350",
        "fontSize": 15,
      },
      Object {},
    ]
  }
  testID=""
>
  (
  <Text>
    Until
  </Text>
   
  <Text>
    Today
  </Text>
  )
</Text>
`;
